File: morgorithm/week9/javascript/solution.js
const solution = (t, p) =>
  new Array(t.length - p.length + 1).fill(0).reduce((acc, _, i) => {
    if (+p >= +t.substring(i, i + p.length)) {
      acc++;
    }
    return acc;
  }, 0);

console.log(solution("3141592", "271") == 2);
console.log(solution("500220839878", "7") == 8);
console.log(solution("10203", "15") == 3);

File: morgorithm/week9/java/Solution.java
package week9.java;

import java.util.stream.IntStream;

public class Solution {

  public int solution(String t, String p) {
    long max = Long.parseLong(p);
    int size = p.length();
    return (int) IntStream.rangeClosed(0, t.length() - p.length())
        .mapToLong((i) -> Long.parseLong(t.substring(i, i + size)))
        .filter((num) -> num <= max)
        .count();
  }

  public static void main(String[] args) {
    System.out.println(
        new Solution().solution("3141592", "271") == 2);
    System.out.println(
        new Solution().solution("500220839878", "7") == 8);
    System.out.println(
        new Solution().solution("10203", "15") == 3);
  }
}
